templates/federated_reporting/mrproper.py

```diff
@@ -0,0 +1,244 @@
+            "roles": ["mrproper"],
+        },
+    )
+    if response["status"] != 201:
+        print("Problem creating mrproper user on superhub. {}".format(response))
+        sys.exit(1)
+
+    response = api.fr_remote_hubs()
+    for hub in response['hubs']:
+        feeder_credentials = getpass(prompt="Enter admin credentials for {} at {}".format(hub['ui_name'], hub['api_url']))
+        feederHostname = hub["ui_name"]
+        feederApi = NovaApi(
+            api_user="admin",
+            api_password=feeder_credentials,
+            cert_path=CERT_PATH,
+            hostname=feederHostname,
+        )
+
+        print("Creating mrproper role on {}".format(feederHostname))
+        response = feederApi.put(
+            "role",
+            "mrproper",
+            {
+                "description": "mrproper Federated Host Cleanup role",
+                "includeContext": "cfengine",
+            },
+        )
+        if response["status"] != 201:
+            print("Problem creating mrproper role on superhub. {}".format(response))
+            sys.exit(1)
+        response = feederApi.put_role_permissions("mrproper", ["host.delete"])
+        if response["status"] != 201:
+            print("Unable to set RBAC permissions on role mrproper")
+            sys.exit(1)
+        print("Creating mrproper user on {}".format(feederHostname))
+        response = feederApi.put(
+            "user",
+            "mrproper",
+            {
+                "description": "mrproper Federated Host Cleanup user",
+                "email": "{}".format(email),
+                "password": "{}".format(mrproper_password),
+                "roles": ["mrproper"],
+            },
+        )
+        if response["status"] != 201:
+            print("Problem creating mrproper user on {}. {}".format(feederHostname, response))
+            sys.exit(1)
+        write_secret(MRPROPER_SECRET_PATH, mrproper_password)
+
+
+def main():
+    if not os.path.exists(MRPROPER_SECRET_PATH):
+        if sys.stdout.isatty():
+            interactive_setup()
+        else:
+            print(
+                "{} requires manual setup, please run as root interactively.".format(
+                    os.path.basename(__file__)
+                )
+            )
+            sys.exit(1)
+
+    mrproper_password = read_secret(MRPROPER_SECRET_PATH)
+    api = NovaApi(
+        api_user="mrproper", api_password=mrproper_password
+    )  # defaults to localhost
+    response = api.fr_hub_status()
+    if not (
+        response["status"] == 200
+        and response["role"] == "superhub"
+        and response["configured"]
+    ):
+        print(
+            "{} can only be run on a Federated Reporting hub configured to be superhub".format(
+                os.path.basename(__file__)
+            )
+        )
+        sys.exit(1)
+
+    response = api.fr_remote_hubs()
+    for hub in response['hubs']:
+        if hub["role"] != "feeder":
```

Comment:
I had the thought that we should not process feeders which are paused, so need to add code here to continue in that case as well. That way we don't delete hostkeys on a paused feeder, because the feeder is not providing any updated information about the hostkey in question.